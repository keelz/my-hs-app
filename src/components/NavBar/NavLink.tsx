import * as React from 'react';

export interface NavLinkProps {
  render?: (children: React.ReactNode) => JSX.Element;
}

const NavLink: React.SFC<NavLinkProps> = props =>
  <div className="Nav-link">
    { props.render
        ? props.render(props.children)
        : props.children
    }
  </div>;

export default NavLink;
