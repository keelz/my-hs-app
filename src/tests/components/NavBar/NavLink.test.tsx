import * as React from 'react';
import * as enzyme from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import NavLink from '../../../components/NavBar/NavLink';

describe('NavBar NavLink', () => {
  describe('snapshots', () => {
    it('renders without crashing', () => {
      const tree = enzyme.shallow(<NavLink />);
      expect(shallowToJson(tree)).toMatchSnapshot();
    });

    it('renders with children correctly', () => {
      const Component = () =>
        <NavLink><div>test</div></NavLink>;
      const tree = enzyme.shallow(<Component />);
      expect(shallowToJson(tree)).toMatchSnapshot();
    });
  });

  describe('integration', () => {
    it('renders with a render prop correctly', () => {
      const render = jest.fn();
      enzyme.shallow(<NavLink render={render} />);
      expect(render).toHaveBeenCalled();
    });
  });
});
