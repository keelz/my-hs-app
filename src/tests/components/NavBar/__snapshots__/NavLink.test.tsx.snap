// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar NavLink snapshots renders with children correctly 1`] = `
<NavLink>
  <div>
    test
  </div>
</NavLink>
`;

exports[`NavBar NavLink snapshots renders without crashing 1`] = `
<div
  className="Nav-link"
/>
`;
